pyxdf/pyxdf.py

```diff
@@ -210,7 +214,11 @@ def load_xdf(filename,
     if not os.path.exists(filename):
         raise Exception('file %s does not exist.' % filename)
 
-    # dict of returned streams, in order of apparance, indexed by stream id
+    # load_only contains the streams to load
+    if isinstance(load_only, int):
```

Comment:
Should check if not string and not iterable. I don't think we have to be bound to StreamIDs being ints.

Reply:
The StreamID is the only value guaranteed to be unique for a stream, and a StreamID is always an integer. I don't understand why we should also handle a string - could you elaborate please?

Reply:
Python doesn't enforce typing so users can pass anything into the load_only argument. Here you are doing a bit of type-checking (among types that are implicitly allowed). Why not go all the way and do full type-checking?

Your line of code is basically saying "if it's not an iterable make it an iterable". I was proposing to explicitly check if it was an iterable. Unfortunately, `str` objects are also considered iterable, so usually you want to exclude those from the check.

After you make sure the argument is a proper iterable, you can then check that each item is an int.

This kind of type-checking is a bit of an anti-pattern in Python and it's considered better-practice to do duck-typing (try, except, else), but I've never been too concerned with that and it was a bigger change than what I was proposing here.